### ISSUE TITLE
Implement edit of transfer transactions

### ISSUE THREAD

**Issue by nazrinharris** [nazrinharris/pecunia#71]
yes pls.

## `TransferTxnBottomSheet` improvement
- [x] Create a `getTransactionById` method (because I never had one??)
- [x] Update the sheet to show both transactions
- [x] Handle the possible errors from the two sources
- [x] Update the `ExpandableTransferDetailsMetadata` to clearly show that `account_name` isn't something that's stored in `TransferDetails`, heck, even removing it sounds like a swell idea

## Deedos
- [x] Implement an `EditTransferTxnFormWidget`
- [x] Update the edit callback to use it

## Possible bugs
- [x] #79

## Current bugs
- [x] The account balance doesn't seem to be updating correctly, try replicating this in tests and see what the issue is. (it was because I was deleting the new transaction, so the amount uses the new transaction, it should've used the old one)

**Comment by nazrinharris** [comment]
33583ac is pretty interesting because it's the first time I'm merging two `AsyncValue`s. The current solution is technically a workaround until Remi puts out an actual `AsyncMerger` of sorts, but as of now, it works like such:

1. Using the `.wait` extension on records, I can wait for multiple futures, like so:
```dart
@riverpod
Future<(Account account, Transaction txn)> getLinkedAccountAndTxn(
  GetLinkedAccountAndTxnRef ref, {
  required String accountId,
  required String txnId,
  bool? debugReturnErrorAccount,
  bool? debugReturnErrorTransaction,
}) async {
  final (Account account, Transaction txn) = await (
    ref.watch(getAccountByIdProvider(accountId, debugReturnError: debugReturnErrorAccount).future),
    ref.watch(getTransactionByIdProvider(txnId, debugReturnError: debugReturnErrorTransaction).future)
  ).wait;
  return (account, txn);
}
```
2. But this brings about a problem, how would I pattern match the errors? Well, you can visit the docs [here](https://api.flutter.dev/flutter/dart-async/ParallelWaitError-class.html) for a more in-depth explanation, but with that, I can pattern match like so:
```dart
return switch (linkedAccountAndTxnValue) {
      AsyncLoading() => Column(
          children: [
            if (enableTopDivider) Divider(color: Colors.grey.withOpacity(0.1)),
            const ListTile(title: CupertinoActivityIndicator()),
            if (enableBottomDivider) Divider(color: Colors.grey.withOpacity(0.1)),
          ],
        ),
      AsyncData(value: (final Account linkedAccount, final Transaction linkedTransaction)) =>
        BuildTransferTxnContent(
          txn: txn,
          linkedTxn: linkedTransaction,
          account: account,
          linkedAccount: linkedAccount,
          onTap: onTap,
          enableTopDivider: enableTopDivider,
          enableBottomDivider: enableBottomDivider,
        ),
      AsyncError(
        error: ParallelWaitError(
          errors: (final a.AsyncError? accountError, final a.AsyncError? transactionsError),
          values: (final Account? _, final Transaction? _),
        ),
        stackTrace: _
      ) =>
        BuildTransferTxnError(
          error: (accountError ?? transactionsError!).error as Failure,
          txn: txn,
          onTap: onTap,
          enableTopDivider: enableTopDivider,
          enableBottomDivider: enableBottomDivider,
        ),
      _ => const Center(child: Text('Something went wrong')),
    };
```
To put it simply, if there's an error in either futures, `wait` will return a `ParallelWaitError`, with two fields, `values` and `errors`. If the future completes, it's kept in the predefined position of in `values`, while its correspondence in `errors` is `null`, and vice versa.

**Comment by nazrinharris** [comment]
Should be completed with b05023d in #37
